Merge Code-Log-Js to main

jochshen opened the pull request:
## Pull Request
Merging Code-Log-Js to main 
**Description**
- added javascript to the code-log page 
   - saves code logs to local storage 
   - can edit each code log if you double click 
   - can delete each code log (*need to change delete icon in future branches*)
- made some edits to small CSS and HTML changes 
**Type of change**
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

**Related Issues**
- fixes #69 
- fixes #59 
- fixes #4 
- fixes #85 
- fixes #86 
- fixes #78 
**Screenshots**
![img](https://github.com/cse110-sp24-group8/cse110-sp24-group8/assets/86510539/6fa7bf3e-ac8b-4158-a363-223eb21524be)

jochshen committed:
changed path in static.yml

b1luu committed:
javascript add log functionality

ulises0516 committed:
minor changes to css of code log and fixing of the markdown bug

jochshen committed:
added scrolling in code-log page, saves to local storage, need to fix sidebar closing with more tasks

b1luu committed:
Added overlay element for Js only

Need to update CSS and HTML as well but will do after I address add log functionality.

b1luu committed:
Fixed visual bug when trying to add more than one log by implementing a .hidden selector

b1luu committed:
Added an 'empty' placeholder when User does add log but for blank entries.

b1luu committed:
Updated CSS

b1luu committed:
CSS Update

b1luu committed:
Updated CSS

b1luu committed:
updated CSS

jochshen committed:
Merge branch 'main' of https://github.com/cse110-sp24-group8/cse110-sp24-group8 into main

ulises0516 committed:
Revert "updated CSS"

This reverts commit f51ff25924e889930d744fcb4771f05285d97bc0.

ulises0516 committed:
Revert "Updated CSS"

This reverts commit 28d8ec901245ca31c03227aa31d464c5be0b6488.

ulises0516 committed:
Revert "CSS Update back to joshes last commit was not working with latest commit changes"

This reverts commit eb732b4d3aac358289ddf75e7cec3e448ec6b639.

ulises0516 committed:
Revert "Updated CSS trying to revert to a working code log state"

This reverts commit 590c0f5f7eef08ba63307143bace92295f2a2243.

ulises0516 committed:
Revert "keep going back:"

This reverts commit cfbd6ce6fc40356306aa17be372af020de3290ce.

ulises0516 committed:
Revert "reverting"

This reverts commit 8930203ce32775c8a290cf21763f2583fae17e1b.

ulises0516 committed:
Revert "stop reverting"

This reverts commit bb25192ee3a8207354a1488108c72f9975c7b1d8.

ulises0516 committed:
please

ulises0516 committed:
Pushing brandons changes which redesigned code log page

erhuang623 committed:
Update main.css

jochshen committed:
added scrolling within each text entry, added delete button within each entry, added double clicking editing functionality within each entry

jochshen committed:
Merge branch 'main' of https://github.com/cse110-sp24-group8/cse110-sp24-group8 into main

jochshen committed:
merged changes with main to make pr more smooth with less conflict

b1luu committed:
Fixed the overlay issue when trying to add more than one log.

When you try and add more than one log, then the elements will shift right, off-center the add log pop-up, this bug is now resolved.

b1luu committed:
Merge branch 'code-log-js' of https://github.com/cse110-sp24-group8/cse110-sp24-group8 into code-log-js

b1luu committed:
added the delete button icon change for consistency across features.

b1luu commented:
Fixed the overlay bug for code-log-js 
Description of bug: When you try and add another element, it will cause the pre-existing elements to shift right and disorient the addlog model.

Changed the delete button to the icon of feedback for consistency across feature UI.

KaustubhPaliwal commented on src/scripts/main/public/styles/code-log.css:
```diff
@@ -242,8 +242,123 @@ button.frame {
+}
+
+/* Responsive Styles */
+@media (max-width: 768px) {
+    .container {
+        flex-direction: column;
+        margin: 10px;
+        width: calc(100% - 20px);
+    }
+
+    .feedbacklist {
+        width: 100%;
+        max-width: none;
+    }
+
+    .modal-content {
+        width: 90%;
+        height: auto;
+    }
+
+    .log-entry {
+        width: calc(100% - 20px); /* Full width for mobile view */
+    }
+
+    .fieldD2 {
+        height: auto; /* Adjust height for mobile view */
+        overflow-y: scroll;
+    }
+}
```
I think you are trying to adapt the Responsive CSS from the Feedback Page, could you change the HTML identifier to the correct corresponding CodeLog HTML identifiers, so that it responds correctly to mobile screen sizes.

Can you also fix the add button on Mobile devices? (You can add a percentage amount of padding to the button, can look at `feedback.css`

This is what is currently happening:
![image](https://github.com/cse110-sp24-group8/cse110-sp24-group8/assets/122563165/83c7c925-40dc-4650-8e35-9701bed411da)

KaustubhPaliwal reviewed (dismissed):
I think you are trying to adapt the Responsive CSS from the Feedback Page, could you change the HTML identifier to the correct corresponding CodeLog HTML identifiers, so that it responds correctly to mobile screen sizes.

Can you also fix the add button on Mobile devices? (You can add a percentage amount of padding to the button, can look at `feedback.css`

This is what is currently happening:
![image](https://github.com/cse110-sp24-group8/cse110-sp24-group8/assets/122563165/83c7c925-40dc-4650-8e35-9701bed411da)

b1luu committed:
Fixed the delete button to be constantly top right of each dynamically created element, and changed the tags for mobile view.

b1luu committed:
Updated CSS to match feedback

b1luu committed:
Updated tag

b1luu committed:
Changed height of log-entry and width to match feedback

height is capped around 400px, but sure why will fix. And also mobile view for small screens for code-log bug will fix as well.

jochshen committed:
added resizing for code entries whenever it reaches certain margin

jochshen committed:
fixed scrolling and reponsiveness issues

nicholas-ngyn committed:
removed scroll problem

nicholas-ngyn committed:
fixed double scrolling problem

jochshen commented:
Fixed some of the styling errors including: 
- code-entries should expand when margin reach a certain minimized length
- scroll is enabled and smooth whether in fullscreen or window is minimized 
- added padding to the bottom so code-logs don't get cut off

ulises0516 reviewed (dismissed):
This was a great fix the css changes to the code logs looks very clean. Also the annoying bug of the scrolling feature being fixed is a great relief! The local storage being added is a great feature as well. This adds much needed features and gives the code log page a much needed refreshing look.

cheung-arthur reviewed (dismissed):
Thank you for finally fixing the scrolling issue with the code-log page! Perhaps we can consider adjusting sizing of each code log element. Also, instead of making each code log element a div, we can also consider making it a web component using a template.

KaustubhPaliwal committed:
Fixed weird gaps on mobile, modal page not scaling properly, items not organized as columns, and a minor background color in Feedback

KaustubhPaliwal reviewed (approved):
The changes in the styling should work great and help maintain consistency across the app on all devices!

nicholas-ngyn reviewed (approved):
Just checked out the branch and everything seems to be working as intended both on big and small screen sizes. The scrolling is working normally but a thing to fix later on is the spacing of the logs as it seems to be more on the left side.
